File: backend/TheHunterApi/Models/SystemPromptResult.cs
namespace TheHunterApi.Models;

/// <summary>תוצאת getLatestPrompt — טקסט, גרסה, האם fallback מוטבע.</summary>
public class SystemPromptResult
{
    public string Text { get; set; } = "";
    public string Version { get; set; } = "";
    public bool IsFallback { get; set; }
}

File: lib/models/system_prompt_result.dart
/// תוצאת getLatestPrompt — טקסט, גרסה, האם fallback מוטבע.
class SystemPromptResult {
  final String text;
  final String version;
  final bool isFallback;

  const SystemPromptResult({
    required this.text,
    required this.version,
    this.isFallback = false,
  });

  factory SystemPromptResult.fromJson(Map<String, dynamic> json) {
    return SystemPromptResult(
      text: json['text'] as String? ?? '',
      version: json['version'] as String? ?? '',
      isFallback: json['isFallback'] as bool? ?? false,
    );
  }
}
